### DISCUSSION TITLE
receive_bytes returns bytearray, not bytes?

### DISCUSSION THREAD

**Discussion by davidbrochart** [frankie567/httpx-ws#38] [Q&A]
It looks like `receive_bytes` actually returns a `bytearray`, not `bytes`? Not sure it makes a lot of difference though.

**Comment by frankie567** [comment]
I tried with the following server and client:

```py
from starlette.applications import Starlette
from starlette.routing import WebSocketRoute


async def ws_hello(websocket):
    await websocket.accept()
    message = await websocket.receive_bytes()
    print(type(message))
    await websocket.close()


app = Starlette(
    routes=[
        WebSocketRoute("/ws", ws_hello),
    ],
)
```

```py
import httpx
from httpx_ws import connect_ws
from httpx_ws.transport import ASGIWebSocketTransport

with connect_ws("http://localhost:8000/ws") as ws:
    ws.send_bytes(b"Hello")
```

The type of `message` is `bytes`, as expected.

Do you have a reproducible example of this behavior?

**Reply by davidbrochart** [reply]
Thanks for your example, it's almost that but the other way around:
```py
import asyncio
from starlette.applications import Starlette
from starlette.routing import WebSocketRoute


async def ws_hello(websocket):
    await websocket.accept()
    await websocket.send_bytes(b"Hello")
    await asyncio.sleep(1)
    await websocket.close()


app = Starlette(
    routes=[
        WebSocketRoute("/ws", ws_hello),
    ],
)
```
```py
import httpx
from httpx_ws import connect_ws
from httpx_ws.transport import ASGIWebSocketTransport

with connect_ws("http://localhost:8000/ws") as ws:
    message = ws.receive_bytes()
    print(type(message))
```
The type of `message` is `bytearray`.

**Reply by frankie567** [reply]
Thank you for the example, very helpful.

So indeed, we get a `bytearray`. It's caused by the underlying implementation of `wsproto` whiches uses a `bytearray`: https://github.com/python-hyper/wsproto/blob/c0a107939d6c0fccdb55028b39b3db026319b65e/src/wsproto/frame_protocol.py#L255

The thing is they define the type hint of a `BytesMessage` as `bytes`: https://github.com/python-hyper/wsproto/blob/c0a107939d6c0fccdb55028b39b3db026319b65e/src/wsproto/events.py#L257

I'll open an issue/discussion on their side about this.

---

That said, apparently, `bytes` and `bytearray` are almost the same, the only difference is that `bytearray` is mutable: https://stackoverflow.com/a/62903378/3864300

**Reply by frankie567** [reply]
Issue on `wsproto` -> https://github.com/python-hyper/wsproto/issues/185
